cmd/crc/cmd/root.go

```diff
@@ -154,3 +160,15 @@ func newMachine() machine.Client {
 func newMachineWithConfig(config crcConfig.Storage) machine.Client {
 	return machine.NewClient(constants.DefaultName, isDebugLog(), network.ParseMode(config.Get(cmdConfig.NetworkMode).AsString()))
 }
+
+func segmentMiddleware(input func(cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) error {
```

Comment:
As mentioned previosuly. This wrapper can be generalized `executeWithLogging` or whatever and then use 

```
if serr := segmentClient.Upload(cmd.Use, err); serr != nil {
```

also clearly behind a `consent` flag here. This is not clear from first glance.

Reply:
segmentClient is built with a config object. `segmentClient, err = segment.NewClient(config)`